sky/jobs/utils.py

```diff
@@ -424,36 +473,24 @@ def cancel_job_by_name(job_name: str) -> str:
 
 def stream_logs_by_id(job_id: int, follow: bool = True) -> str:
     """Stream logs by job id."""
-    controller_status = job_lib.get_status(job_id)
-    status_msg = ux_utils.spinner_message(
-        'Waiting for controller process to be RUNNING') + '{status_str}'
-    status_display = rich_utils.safe_status(status_msg.format(status_str=''))
+
+    def should_keep_logging(status: managed_job_state.ManagedJobStatus) -> bool:
```

Comment:
minor: Should we allow this function to take `Optional[managed_job_state.ManagedJobStatus`, so that we don't have to assert before every invocation of this function?

Reply:
I don't really like any of the alternatives:
1. Function accepts `Optional[managed_job_state.ManagedJobStatus]` but throws if it gets None - feels like the type is just wrong in that case.
2. Function accepts `Optional[managed_job_state.ManagedJobStatus]` and returns True (or maybe False?) if it gets None - feels inaccurate.
3. Wrap `get_latest_task_id_status` to assert non-None status - feels overengineered.
4. Change all the status-fetching functions to throw if the job does not exist - maybe correct, but out of scope for this PR.

So I guess I'd prefer to just keep this - it's a bit verbose but it's pretty simple and understandable.